src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -1102,19 +1105,19 @@ public override string ToString()
         }
 
         /// <summary>
-        /// Convert the <see cref="DownloadFrom"/> and <see cref="DownloadTo"/> to a list of consecutive hashes where the fist item in the list is the hash of the download from.
+        /// Convert the <see cref="DownloadFrom"/> and <see cref="DownloadTo"/> to a list of consecutive hashes where the first item in the list is the hash of <see cref="DownloadFrom"/>.
         /// </summary>
-        public uint256[] ToHashList()
+        public ChainedHeader[] ToHashArray()
         {
             int itemsCount = this.DownloadTo.Height == this.DownloadFrom.Height ? 1 : this.DownloadTo.Height - this.DownloadFrom.Height + 1;
 
-            var hashes = new uint256[itemsCount];
+            var hashes = new ChainedHeader[itemsCount];
 
             ChainedHeader currentHeader = this.DownloadTo;
 
             for (int i = hashes.Length - 1; i >= 0; --i)
```

Comment:
I would use `i--` for consistency with other loops in the code.

Reply:
Is it not actually wrong, this will decrement i before it enters the loop

Reply:
No, it does exactly the same as postfix version, the only problem with it is inconsistency with other code.